core/cautils/fileutils.go

```diff
@@ -325,15 +326,14 @@ func GetFileFormat(filePath string) FileFormat {
 }
 
 // handleListObject handle a List manifest
-func handleListObject(obj workloadinterface.IMetadata) []workloadinterface.IMetadata {
+func handleListObject(list *listworkloads.ListWorkloads) []workloadinterface.IMetadata {
```

Comment:
question: why do we need to change the `obj` argument to a `list` argument of concrete type?

If we do this, from now on client code would have to always assert the type of the passed object. Not to mention that this sort of violates the Dependency Inversion principle of SOLID: we would now depend on a concrete type, and not an interface.

Is there a reason why we need to accept a concrete type?

Reply:
@Moshe-Rappaport-CA  What Vlad is proposing here is that you should add an `IList`  interface so the `Lisy` struct will be compatible with it, this way you support an interface and not an object.